beets/importer.py

```diff
@@ -572,18 +572,22 @@ def remove_duplicates(self, lib):
                 util.prune_dirs(os.path.dirname(item.path),
                                 lib.directory)
 
-    def set_fields(self):
+    def set_fields(self, lib):
         """Sets the fields given at CLI or configuration to the specified
-        values.
+        values, for both the album and all its items.
         """
-        for field, view in config['import']['set_fields'].items():
-            value = view.get()
-            log.debug(u'Set field {1}={2} for {0}',
-                      displayable_path(self.paths),
-                      field,
-                      value)
-            self.album[field] = value
-        self.album.store()
+        with lib.transaction():
+            for field, view in config['import']['set_fields'].items():
+                value = view.get()
+                log.debug(u'Set field {1}={2} for {0}',
+                          displayable_path(self.paths),
+                          field,
+                          value)
+                self.album[field] = value
+                for item in self.imported_items():
```

Comment:
This loop occurs for each iteration of `for field, view in ...`. Therefore, `self.imported_items()` should probably be 'materialized' once before the outer loop. (iirc, `self.imported_items()` returns a list of 'unmaterialized' DB results.) I'm not sure, however, how this materialization should be implemented.

Reply:
done